src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -139,23 +138,7 @@ export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
       expiration: futureDate,
       noExpiration: false,
     });
-  }, [visible]); // eslint-disable-line react-hooks/exhaustive-deps
-
-  useEffect(() => {
-    if (awsRegions && awsRegions.length) {
-      dispatch({
-        type: "editAWSRegion",
-        region: userAwsRegion || awsRegions[0],
-      });
-    }
-    if (publicKeys && publicKeys.length) {
-      dispatch({
-        type: "editPublicKey",
-        publicKey: publicKeys[0],
-        savePublicKey: false,
-      });
-    }
-  }, [awsRegions, userAwsRegion, publicKeys, dispatch]);
+  }, [visible, userAwsRegion, awsRegions, publicKeys]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I deleted the hook below and combined it with this one here. The information was pretty much the same so I figured it would be okay

(The deleted hook was introduced in evergreen-ci/spruce/pull/525 and appears to be specifically for spawning hosts from tasks. However I didn't really notice any difference after combining them.)